lib/configs/recommended.js

```diff
@@ -1,113 +1,113 @@
-    'no-useless-escape': 'error',
-    'no-useless-return': 'error',
-    'no-unused-expressions': 'error',
-    'no-void': 'error',
-    'no-with': 'error',
-    radix: 'error',
-    'vars-on-top': 'error',
-    'wrap-iife': ['error', 'any'],
+    },
+    rules: {
+      // Possible errors
+      // https://eslint.org/docs/rules/#possible-errors
+      'no-console': 'error',
 
-    // Variables
-    // https://eslint.org/docs/rules/#variables
-    'no-label-var': 'error',
-    'no-shadow': 'error',
-    'no-shadow-restricted-names': 'error',
-    'no-undef-init': 'error',
-    'no-unused-vars': ['error', { vars: 'all', args: 'after-used' }],
-    'no-use-before-define': ['error', { functions: false }],
+      // Best practices
+      // https://eslint.org/docs/rules/#best-practices
+      'array-callback-return': 'error',
+      'consistent-return': 'error',
+      'default-case': 'error',
+      'dot-notation': ['error', { allowKeywords: true }],
+      eqeqeq: ['error', 'smart'],
+      'guard-for-in': 'error',
+      'no-alert': 'error',
+      'no-caller': 'error',
+      'no-else-return': 'error',
+      'no-empty-function': [
+        'error',
+        {
+          allow: ['arrowFunctions', 'functions', 'methods'],
+        },
+      ],
+      'no-eval': 'error',
+      'no-extend-native': 'error',
+      'no-extra-bind': 'error',
+      'no-floating-decimal': 'error',
+      'no-implied-eval': 'error',
+      'no-iterator': 'error',
+      'no-labels': 'error',
+      'no-loop-func': 'error',
+      'no-multi-str': 'error',
+      'no-new': 'error',
+      'no-new-func': 'error',
+      'no-new-object': 'error',
+      'no-new-wrappers': 'error',
+      'no-octal-escape': 'error',
+      'no-proto': 'error',
+      'no-return-assign': 'error',
+      'no-return-await': 'error',
+      'no-script-url': 'error',
+      'no-self-compare': 'error',
+      'no-sequences': 'error',
+      'no-throw-literal': 'error',
+      'no-useless-concat': 'error',
+      'no-useless-escape': 'error',
+      'no-useless-return': 'error',
+      'no-unused-expressions': 'error',
+      'no-void': 'error',
+      'no-with': 'error',
+      radix: 'error',
+      'vars-on-top': 'error',
+      'wrap-iife': ['error', 'any'],
 
-    // Aura specific rules
-    '@salesforce/aura/aura-api': 'error',
-    '@salesforce/aura/getevt-markup-prefix': 'error',
-    '@salesforce/aura/no-deprecated-aura-error': 'error',
-    '@salesforce/aura/no-deprecated-component-creation': 'error',
-  },
+      // Variables
+      // https://eslint.org/docs/rules/#variables
+      'no-label-var': 'error',
+      'no-shadow': 'error',
+      'no-shadow-restricted-names': 'error',
+      'no-undef-init': 'error',
+      'no-unused-vars': ['error', { vars: 'all', args: 'after-used' }],
+      'no-use-before-define': ['error', { functions: false }],
+
+      // Rules that have been enabled by default in eslint v9 but disabled to keep the config backward compatible
```

Comment:
I have mixed feelings about this: do we really need to keep things backward compatible since we're already introducing a breaking change with the config file format change?
I feel that this upgrade is probably the best time to re-align with the latest "standard/recommended" ESLint config.

I would remove these rule overrides from the recommended config and add a "migration" section in the readme that lists those rules so that users can disable them manually if they want to migrate with the least effort.